javaslang/src/main/java/javaslang/collection/Collections.java

```diff
@@ -22,6 +22,78 @@
  */
 final class Collections {
 
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    static boolean equals(Map<?, ?> map1, Object object) {
+        if (map1 == null ^ object == null) {
```

Comment:
@v1ctor allright, then we will do the check like this:

```java
@SuppressWarnings("unchecked")
static <K, V> boolean equals(Map<K, V> source, Object o) {
    if (o == source) {
        return true;
    } else if (source != null && o instanceof Map) {
        final Map<???, ???> that = (Map<???, ???>) o;
        return source.size() == that.size() && ???;
    } else {
        return false;
    }
}
```

I will investigate now how to deal with the sorted collections (as mentioned before).